lib/language_pack/ruby.rb

```diff
@@ -219,34 +219,46 @@ def ruby_version
   # default JAVA_OPTS
   # return [String] string of JAVA_OPTS
   def default_java_opts
-    "-Xss512k -XX:+UseCompressedOops -Dfile.encoding=UTF-8"
+    '-Dfile.encoding=UTF-8'
+  end
+
+  def set_jvm_stack_size
+    ENV['JVM_STACK_SIZE'] = env('JVM_STACK_SIZE') || 512
   end
 
   def set_jvm_max_heap
     <<-EOF
 case $(ulimit -u) in
-256)   # 1X Dyno
-  JVM_MAX_HEAP=384
+512)   # 2X, private-s: memory.limit_in_bytes=1073741824
+  JVM_MAX_HEAP=671
   ;;
-512)   # 2X Dyno
-  JVM_MAX_HEAP=768
-  ;;
-16384) # IX Dyno
+16384) # perf-m, private-m: memory.limit_in_bytes=2684354560
   JVM_MAX_HEAP=2048
   ;;
-32768) # PX Dyno
-  JVM_MAX_HEAP=5120
+32768) # perf-l, private-l: memory.limit_in_bytes=15032385536
+  JVM_MAX_HEAP=12288
+  ;;
+*) # Free, Hobby, 1X: memory.limit_in_bytes=536870912
+  JVM_MAX_HEAP=300
   ;;
 esac
-EOF
+    EOF
+  end
+
+  def set_java_stack_size
+    <<-EOF
+if ! [[ "${JAVA_OPTS}" == *-Xss* ]]; then
+  export JAVA_STACK=${JAVA_STACK:--Xss${JVM_STACK_SIZE:-512}k}
```

Comment:
should this be `export JAVA_OPTS=${JAVA_OPTS:--Xss${JVM_STACK_SIZE:-512}k}`?

I don't see where `JAVA_STACK` (and thus the -Xss` option) gets added to `JAVA_OPTS` or the command.

Reply:
Updated, sorry this file bamboozles me, probably need to refactor it